extensions/amp-story/1.0/amp-story.js

```diff
@@ -1284,17 +1339,57 @@ export class AmpStory extends AMP.BaseElement {
       return Promise.resolve();
     }
 
-    if (
-      isExperimentOn(this.win, 'amp-story-paywall-exp') &&
-      targetPage.isPaywallProtected()
-    ) {
-      if (this.storeService_.get(StateProperty.SUBSCRIPTIONS_DIALOG_STATE)) {
-        // Subscription dialog is already triggered.
-        return Promise.resolve();
-      }
-      this.storeService_.dispatch(Action.TOGGLE_SUBSCRIPTIONS_DIALOG, true);
+    const subscriptionsState = this.storeService_.get(
+      StateProperty.SUBSCRIPTIONS_STATE
+    );
+    if (this.isPaywallStory_()) {
```

Comment:
now that this is like 50 lines, maybe we can also pull some of it into smaller methods for readability?

Reply:
Done.

Reply:
It is super important to only keep one level of indentation (i.e. never a `if` within a `if`), here you're going 5 levels + using `else` statements, making it impossible to follow unless compiling the code in your head.

Reply:
Was trying to keep all paywall-related logic in a single block, but I can also see the advantages of keeping a very flat structure here. I flattened things a bit so that now everything is at most two levels, where every block handles a specific scenario. For the remaining two-level stuff, I am kind of not sure how to flatten it further without duplicate conditions.

Reply:
Made everything one-level indentation by separating out the logic of showing the UI and switch to the paywall page. Thanks Caleb!